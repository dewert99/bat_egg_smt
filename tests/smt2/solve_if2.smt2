(declare-const i Bool)
(declare-const x Bool)
(declare-const y Bool)
(declare-const o Bool)
(assert (or (if i x y) o))
(check-sat-assuming ((not o) (not y)))
(get-value (i x y o))
(get-model)
(declare-sort U)
(declare-const a U)
(declare-const b U)
(assert (= (if i a b) b))
(check-sat-assuming ((not (= a b))))
(get-value (i x y o a b))
(get-model)