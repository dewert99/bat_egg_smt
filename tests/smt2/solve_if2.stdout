sat
((i true)
 (x true)
 (y false)
 (o false))
(
 (define-fun i () Bool true)
 (define-fun o () Bool false)
 (define-fun x () Bool true)
 (define-fun y () Bool false)
)
sat
((i false)
 (x true)
 (y false)
 (o true)
 (a @U_9)
 (b @U_11))
(
 (define-fun a () U @U_9)
 (define-fun b () U @U_11)
 (define-fun i () Bool false)
 (define-fun o () Bool true)
 (define-fun x () Bool true)
 (define-fun y () Bool false)
)
